### PR TITLE
Nvt detailspage test

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* Added tests for NVT Details Pages.
* No permissions as NVTs are Global Objects
* No Copy, Clone, Edit or Delete stuff for NVTs

While there:
* Updating the DateTime format for Note and Override Boxes

**Why**:

* Tests have been missing so far ....

**How**:

**Checklist**:

- [x] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

**Comment by y0urself** [comment]
Finally ...

File: gsa/src/web/entity/__tests__/box.js [hunk 1]
```diff
@@ -17,22 +17,36 @@
...
+import {rendererWith} from 'web/utils/testing';
+import {setTimezone, setUsername} from 'web/store/usersettings/actions';
 
 import EntityBox from '../box';
 
 setLocale('en');
 
-const date = Date('2019-01-01T12:00:00Z');
-const date2 = Date('2019-02-02T12:00:00Z');
+const manualUrl = 'test/';
```
Comment (sarahd93) [inline gsa/src/web/entity/__tests__/box.js:29]: Do we really need the manualUrl here? I can't figure out where that would be used.

Comment (swaterkamp) [reply gsa/src/web/entity/__tests__/box.js:29]: It's used in the help icon to direct to the manual. Normally it's set in GOS, but here we probably need it for testing to be test...

Comment (sarahd93) [reply gsa/src/web/entity/__tests__/box.js:29]: I'm sorry, I should have phrased my question better. This is the test for the entity box. Where do we use a help icon here? I can't find any manual links here and I don't think we need the manualUrl for anything else.

And as a note: for tests with a help icon (or rather with a manual link) the manualUrl needs to be set but it doesn't have to be `'test/'` it just needs to be defined.

File: gsa/src/web/pages/nvts/__tests__/detailspage.js [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+    _id: '',
+    name: '',
+    trash: 0,
+  },
+  text: 'test_override_2',
+  threat: 'Internal Error',
+  writable: 1,
+});
+
+const getNvt = jest.fn().mockResolvedValue({
```
Comment (sarahd93) [inline gsa/src/web/pages/nvts/__tests__/detailspage.js:212]: The globally defined mocks need to be inside a beforeEach now.

File: gsa/src/web/entity/__tests__/box.js [hunk 3]
```diff
@@ -17,22 +17,29 @@
  */
 import React from 'react';
 
-import Date, {setLocale} from 'gmp/models/date';
+import date, {setLocale} from 'gmp/models/date';
 
-import {render} from 'web/utils/testing';
+import {rendererWith} from 'web/utils/testing';
+import {setTimezone} from 'web/store/usersettings/actions';
```
Comment (sarahd93) [inline gsa/src/web/entity/__tests__/box.js:23]:
```suggestion
import {setTimezone} from 'web/store/usersettings/actions';
import {rendererWith} from 'web/utils/testing';
```

File: gsa/src/web/entity/__tests__/note.js [hunk 4]
```diff
@@ -22,6 +22,7 @@ import {setLocale} from 'gmp/models/date';
 import Note from 'gmp/models/note';
 
 import {rendererWith} from 'web/utils/testing';
+import {setTimezone} from 'web/store/usersettings/actions';
```
Comment (sarahd93) [inline gsa/src/web/entity/__tests__/note.js:25]:
```suggestion
import {setTimezone} from 'web/store/usersettings/actions';
import {rendererWith} from 'web/utils/testing';
```

File: gsa/src/web/entity/__tests__/override.js [hunk 5]
```diff
@@ -22,6 +22,7 @@ import {setLocale} from 'gmp/models/date';
 import Override from 'gmp/models/override';
 
 import {rendererWith} from 'web/utils/testing';
+import {setTimezone} from 'web/store/usersettings/actions';
```
Comment (sarahd93) [inline gsa/src/web/entity/__tests__/override.js:25]:
```suggestion
import {setTimezone} from 'web/store/usersettings/actions';
import {rendererWith} from 'web/utils/testing';
```

**Review by sarahd93 — CHANGES_REQUESTED** [review]

File: gsa/src/web/entity/__tests__/note.js (hunk 4 above)
Comment (y0urself) [reply gsa/src/web/entity/__tests__/note.js:25]: Can you explain the ordering of the imports to me @sarahd93 ?

Comment (sarahd93) [reply gsa/src/web/entity/__tests__/note.js:25]: Our coding guide for gsa says "alphabetically within the above categories" so I figured web/store comes before web/utils.

**Review by sarahd93 — APPROVED** [review]

**Pull request merged**